### PR TITLE
View all services on interaction

### PR DESCRIPTION
**Pull request by Davit-BetterAngels**
DEV-845

## Summary by Sourcery

Refactor the service management components by consolidating ProvidedServices and RequestedServices into a single RequestedProvidedServices component. Introduce a ServicesModal for improved service selection and management. Enhance the enum display mapping to include service request types.

Enhancements:
- Refactor the ProvidedServices and RequestedServices components to use a unified RequestedProvidedServices component, simplifying the codebase and reducing redundancy.
- Introduce a ServicesModal component to handle service selection, improving the user interface for managing services.
- Add a new enumDisplayServiceType mapping for better display of service request types.

### CODE REVIEW

**Review by shiozakiyuko — APPROVED** [review]
It looked good to me on iOS and Android. Thanks so much! Approved!

File: libs/expo/shared/ui-components/src/lib/PillContainer/PillContainer.stories.tsx [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+    type: 'success',
+    data: ['Service 1', 'Service 2', 'Service 3', 'Service 4', 'Service 5'],
+  },
+};
+
+export default meta;
+
+type PillContainerStory = StoryObj<typeof PillContainer>;
+
+export const Basic: PillContainerStory = {
```
Comment (tglaz) [inline libs/expo/shared/ui-components/src/lib/PillContainer/PillContainer.stories.tsx:17]: nice! I need to go back and add couple of these myself.

File: libs/expo/shared/ui-components/src/lib/PillContainer/PillContainer.tsx [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+
+export function PillContainer({
+  data,
+  type,
+}: {
+  data: string[];
+  type?: 'primary' | 'success';
+}) {
+  const [showAll, setShowAll] = useState(false);
+  const servicesToDisplay = showAll ? data : data.slice(0, 5);
```
Comment (tglaz) [inline libs/expo/shared/ui-components/src/lib/PillContainer/PillContainer.tsx:16]: I'd move the `5` value out into a prop or constant. 
It's hard coded 3 times here and we should be able to switch this easily

maybe PillContainer could just accept a `maxVisible` prop.

**Review by tglaz — CHANGES_REQUESTED** [review]

File: libs/expo/shared/ui-components/src/lib/PillContainer/PillContainer.tsx (hunk 2 above)
Comment (Davit-BetterAngels) [reply libs/expo/shared/ui-components/src/lib/PillContainer/PillContainer.tsx:16]: great idea. will update it

Comment (Davit-BetterAngels) [reply libs/expo/shared/ui-components/src/lib/PillContainer/PillContainer.tsx:16]: done

**Review by tglaz — APPROVED** [review]

File: libs/expo/shared/ui-components/src/lib/PillContainer/PillContainer.tsx [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+import TextBold from '../TextBold';
+
+export function PillContainer({
+  data,
+  type,
+  maxVisible,
+}: {
+  data: string[];
+  type?: 'primary' | 'success';
+  maxVisible: number;
```
Comment (tglaz) [inline libs/expo/shared/ui-components/src/lib/PillContainer/PillContainer.tsx:15]: nice! 
aprvd. if wanna make 1 more change you can also maxVisible default to a 5, if that'd be our standard. but can do later

**Pull request merged**